utils/estimateDynamicsUsingIntervals.m

```diff
@@ -34,13 +34,33 @@
          for i=1:size(input.ftNames,1)
                     estimatedFtData.(input.ftNames{i})=zeros(size(dataset.ftData.(input.ftNames{i})));
          end
-                newOrdering=true;
+         newOrdering=true;
+         
+         % extract sub-intervals in the case of walking datasets
+         isLeftLegIntervDef = ismember('leftLeg',intervalsNames);
+         isRightLegIntervDef = ismember('rightLeg',intervalsNames);
+         if(isLeftLegIntervDef || isRightLegIntervDef)
+             [~, initStanceLeft, endStanceLeft, initStanceRight, endStanceRight] = getStancePeriodsFromWalkingFTdata(...
+                 dataset.time, dataset.ftData.left_foot,...
+                 dataset.time, dataset.ftData.right_foot,...
+                 input.gait.minSwingLength, input.gait.tolerancePercentage, input.gait.shrinkPercentage);
+             % set the sub-intervals
+             if(isLeftLegIntervDef)
+                 input.intervals.leftLeg=struct('initTime',initStanceLeft,'endTime',endStanceLeft,'contactFrame','l_sole');
+             end
+             if(isRightLegIntervDef)
+                 input.intervals.rightLeg=struct('initTime',initStanceRight,'endTime',endStanceRight,'contactFrame','r_sole');
+             end
+         end
+         
          %   generalize ordering of intervals end first section
```

Comment:
So this is the section that will work for the walking dataset but I do not see how you discriminate from a walking dataset from a non walking dataset. Does this work correctly for the yoga too? and what would happen with a grid dataset?

Reply:
You're right. as it is now, I assume the robot is on the pole, or walking. I see two options for solving this issue:
- have more specific fields in `input.intervals.`: hanging, fixed, **yogaRightLeg, yogaLeftLeg, walkingRightLeg, walkingLefteg**
- use the field `input.type` you describe in https://github.com/robotology-playground/insitu-ft-analysis/blob/master/docs/howToFiles/prepareData.md, to wrap the code I've added (l.39 - l.54).

Second option would be more elegant.

Reply:
implemented in https://github.com/robotology-playground/insitu-ft-analysis/pull/15